Client/mods/deathmatch/logic/rpc/CVehicleRPCs.cpp

```diff
@@ -653,3 +654,25 @@ void CVehicleRPCs::SetVehiclePlateText(CClientEntity* pSourceEntity, NetBitStrea
         }
     }
 }
+
+
+void CVehicleRPCs::SetVehicleNitroActivated(CClientEntity* pSourceEntity, NetBitStreamInterface& bitStream)
+{
+    bool state;
+    if (bitStream.ReadBit(state))
+    {
+        CClientVehicle* pVehicle = m_pVehicleManager->Get(pSourceEntity->GetID());
+        if (pVehicle)
+        {
+            if (pVehicle->IsNitroInstalled())
+            {
+                if (state)
+                    pVehicle->SetNitroLevel(pVehicle->GetNitroLevel() - 1.0001f);
+                else
+                    pVehicle->SetNitroLevel(pVehicle->GetNitroLevel() + 1.0001f);
+               
+            }
+        }
+    }
```

Comment:
early return

Why `GetLevel(0 - 1.0001f)` and not just 0/1?

Reply:
I am not sure but this is how the client code use it maybe something about charing level

```c++
bool CStaticFunctionDefinitions::SetVehicleNitroActivated(CClientEntity& Entity, bool bActivated)
{
    RUN_CHILDREN(SetVehicleNitroActivated(**iter, bActivated))

    if (IS_VEHICLE(&Entity))
    {
        CClientVehicle& pVehicle = static_cast<CClientVehicle&>(Entity);
        if (pVehicle.IsNitroInstalled())
        {
            bool bAlreadyActivated;
            IsVehicleNitroActivated(pVehicle, bAlreadyActivated);

            if (bAlreadyActivated != bActivated)
            {
                if (bActivated)
                    pVehicle.SetNitroLevel(pVehicle.GetNitroLevel() - 1.0001f);
                else
                    pVehicle.SetNitroLevel(pVehicle.GetNitroLevel() + 1.0001f);
            }

            return true;
        }
    }

    return false;
}
```

Reply:
// If nitro level < 0, nitro is activated. (until nitro level reaches -1, at that point it will become 0 and increase instead of decrease)

Reply:
that looks very sketchy

Reply:
well from what I understand setting nitro at certain value will trigger the nitro activation same as same for deactivation
since we calling SetNitroLevel()